Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -110,13 +109,30 @@ namespace AZ
                 m_propertyValues.resize(materialAsset.GetPropertyValues().size());
                 AZ_Assert(m_propertyValues.size() == m_layout->GetPropertyCount(), "The number of properties in this material doesn't match the property layout");
 
-                for (size_t i = 0; i < materialAsset.GetPropertyValues().size(); ++i)
+                // Material Asset property names will only be empty when the cvar r_enableMaterialPropertyNames is false. By default, it is true.
```

Comment:
Think about how your comments will age, as the code changes in the future. For example, what if someone changes the default value? It is very unlikely they will find and update this comment.

For example, you could say something like "Material asset property names are optional, see MaterialAssetBuilderComponent.cpp". This is better because it directs the reader where to look for more info, and it's much less likely that MaterialAssetBuilderComponent.cpp will be renamed. But it could still become a broken link.

... an even better approach might be to make the APIs solid enough that they lead the reader naturally to the right place, and are very unlikely to change. Like I suggest above, consider making the MaterialAsset class solve the problem of "how do we get the property values to align" instead of doing it here. Just by searching, that code will naturally lead the user to read MaterialAssetCreator, and that code will lead the user to MaterialAssetBuilderComponent.

There isn't necessarily an easy answer here, but this kind of comment should trigger your "spidey sense" and see what can be improved.

Reply:
Great point, I would hate to be the one that leaves easily stale comments :p